sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,544 @@
+# Form Recognizer Design Overview
+
+The Form Recognizer client library provides two clients to interact with the service: `FormRecognizerClient` and 
+`CustomFormClient`, which can be imported from the `azure.ai.formrecognizer` namespace. The asynchronous clients 
+can be imported from `azure.ai.formrecognizer.aio`.
+
+`FormRecognizerClient` provides methods for interacting with the receipt and layout models.
+`CustomFormClient` provides the methods for training custom models to analyze forms.
+
+Authentication is achieved by passing an instance of `CognitiveKeyCredential("<api_key>")` to the client,
+or by providing a token credential from `azure.identity` to use Azure Active Directory.
+
+## Receipt and Layout
+
+Receipt and layout are accessed through the `FormRecognizerClient`. The input form or document can be passed as a 
+string url/path to the image, or as a file stream. The SDK will determine content-type and send the appropriate header. 
+
+Both receipt and layout methods return poller objects which are used to get the result.
+The `begin_extract_receipt` method returns a `List[ExtractedReceipt]` with hardcoded receipt fields.
+The `begin_extract_layout` method returns the extracted layouts as a `List[ExtractedLayoutPage]`.
+
+If the keyword argument `include_text_details=True` is passed in, the `elements` attribute will be re-hydrated with the
+OCR result for the particular value/cell referenced by the json pointer.
+
+### Form Recognizer Client
+```python
+from azure.ai.formrecognizer import FormRecognizerClient
+
+client = FormRecognizerClient(endpoint: str, credential: Union[CognitiveKeyCredential, TokenCredential])
+
+client.begin_extract_receipt(form: Union[str, BytesIO], **kwargs) -> LROPoller -> List[ExtractedReceipt]
+
+client.begin_extract_layout(form: Union[str, BytesIO], **kwargs) -> LROPoller -> List[ExtractedLayoutPage]
+```
+
+### Receipt Models
+```python
+class ExtractedReceipt(DictMixin):
+    receipt_items: List[ReceiptItem]
+    merchant_address: FieldValue
+    merchant_name: FieldValue
+    merchant_phone_number: FieldValue
+    receipt_type: FieldValue
+    subtotal: FieldValue
+    tax: FieldValue
+    tip: FieldValue
+    total: FieldValue
+    transaction_date: FieldValue
+    transaction_time: FieldValue
+    page_range: PageRange
+    page_metadata: PageMetadata
```

Comment:
What happens if my receipt spans >1 page?
- Which `PageMetadata` would be stored here? The first page of the receipt?
- How would I access the `PageMetadata`'s for the subsequent pages?

Reply:
Maybe I am not setting up the right sample scenario here, but when I send a receipt that spans more than one page, I get each page of the receipt as a separate item in documentResult: [example](https://github.com/kristapratico/azure-sdk-for-python/blob/fr-json/sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/json-2-page/receipt_2page.json#L5345). This is unlike the result I get from calling a custom supervised method, which will be a single documentResult with page range, e.g. [1, 2].

So I may have represented it wrong if I'm not sending a good multi-page sample to the service, but the idea here would be that we have a `list[ExtractedReceipt]` where each item is a page and each `PageMetadata` would represent the OCR on that page.

Reply:
I think this might be a coincidence b/c each item in `documentResult` has a 1 page span (`[1,1]` and `[2,2]`, respectively). 

What if a single `documentResult` has a `pangeRange` from `[1,2]`? Should `page_metadata` be a list? I don't know if we have an example that reflects this scenario or if it would occur in the wild. @bojunehsu?